src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/CanonicalDefinitionEETypeNode.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    internal sealed class CanonicalDefinitionEETypeNode : EETypeNode
+    {
+        public CanonicalDefinitionEETypeNode(NodeFactory factory, TypeDesc type) : base(factory, type)
+        {
+            Debug.Assert(type.IsCanonicalDefinitionType(CanonicalFormKind.Any));
+        }
+
+        public override bool ShouldSkipEmittingObjectNode(NodeFactory factory) => false;
+        public override bool StaticDependenciesAreComputed => true;
+        public override bool IsShareable => IsTypeNodeShareable(_type);
```

Comment:
I think we need to make `__Canon` and `__UniversalCanon` EETypes return true here so that they get generated into each obj file in multifile compilation and we COMDAT fold them away. The logic in `MultiFileCompilationModuleGroup::ContainsType` returns true for everything that is not an EcmaType (which canon types are not).

Cc @nattress because he's the expert here.